### PR TITLE
Moving sobject-refresh command & automatic sobject refresh code to core

### PR DESCRIPTION
**Pull request by jag-sfdc**
### What does this PR do?
Moves sobject-refresh command & automatic sobject refresh code to core. No functionality change for the end user.

### What issues does this PR fix or reference?
@W-8912259@

### Functionality Before
N/A

### Functionality After
Same as before. No functionality change.

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/index.ts [hunk 1]
```diff
@@ -619,6 +629,21 @@ export async function activate(context: vscode.ExtensionContext) {
   telemetryService.sendExtensionActivationEvent(extensionHRStart);
   console.log('SFDX CLI Extension Activated');
   return api;
```
Comment (rcoringrato-sfdc) [inline packages/salesforcedx-vscode-core/src/index.ts:631]: @jag-sfdc Is this return in the proper place?

Comment (jag-sfdc) [reply packages/salesforcedx-vscode-core/src/index.ts:631]: @rcoringrato-sfdc - Definitely not. Fixing it now.

**Review by rcoringrato-sfdc — APPROVED** [review]

File: packages/salesforcedx-vscode-apex/package.json [hunk 2]
```diff
@@ -157,10 +157,6 @@
           "command": "sfdx.force.apex.test.run",
```
Comment (lcampos) [inline packages/salesforcedx-vscode-apex/package.json:157]: You should be able to also remove `salesforce/salesforcedx-sobjects-faux-generator` as a dependency in this extension

Comment (jag-sfdc) [reply packages/salesforcedx-vscode-apex/package.json:157]: Sure, will do that.

**Pull request merged**